sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy

```diff
@@ -1423,4 +1427,170 @@ class ContainerAPITest extends APISpec {
         then:
         thrown(IllegalArgumentException)
     }
+
+    def "Restore Container"() {
+        given:
+        def cc1 = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        cc1.create()
+        def blobName = generateBlobName()
+        cc1.getBlobClient(blobName).upload(defaultInputStream.get(), 7)
+        cc1.delete()
+        def blobContainerItem = primaryBlobServiceClient.listBlobContainers(
+            new ListBlobContainersOptions()
+                .setPrefix(cc1.getBlobContainerName())
+                .setDetails(new BlobContainerListDetails().setRetrieveDeleted(true)),
+            null).first()
+
+        if (!playbackMode()) {
+            Thread.sleep(30000)
+        }
+
+        when:
+        cc1.restore(blobContainerItem.getName(), blobContainerItem.getVersion())
+
+        then:
+        cc1.listBlobs().size() == 1
+        cc1.listBlobs().first().getName() == blobName
+    }
+
+    def "Restore Container into other container"() {
+        given:
+        def cc1 = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        cc1.create()
+        def blobName = generateBlobName()
+        cc1.getBlobClient(blobName).upload(defaultInputStream.get(), 7)
+        cc1.delete()
+        def blobContainerItem = primaryBlobServiceClient.listBlobContainers(
+            new ListBlobContainersOptions()
+                .setPrefix(cc1.getBlobContainerName())
+                .setDetails(new BlobContainerListDetails().setRetrieveDeleted(true)),
+            null).first()
+
+        if (!playbackMode()) {
+            Thread.sleep(30000)
+        }
+
+        when:
+        def cc2 = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        cc2.restore(blobContainerItem.getName(), blobContainerItem.getVersion())
+
+        then:
+        cc2.listBlobs().size() == 1
+        cc2.listBlobs().first().getName() == blobName
+    }
+
+    def "Restore Container with response"() {
+        given:
+        def cc1 = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        cc1.create()
+        def blobName = generateBlobName()
+        cc1.getBlobClient(blobName).upload(defaultInputStream.get(), 7)
+        cc1.delete()
+        def blobContainerItem = primaryBlobServiceClient.listBlobContainers(
+            new ListBlobContainersOptions()
+                .setPrefix(cc1.getBlobContainerName())
+                .setDetails(new BlobContainerListDetails().setRetrieveDeleted(true)),
+            null).first()
+
+        if (!playbackMode()) {
+            Thread.sleep(30000)
+        }
+
+        when:
+        def response = cc1.restoreWithResponse(blobContainerItem.getName(), blobContainerItem.getVersion(),
+            Duration.ofMinutes(1), Context.NONE)
+
+        then:
+        response != null
```

Comment:
It would be good to validate some of the values on the response if there are any to make sure they're being set correctly.

Reply:
ok.